alexandrainst/AIAI-eval
# Tests/add basic tests for task module

This PR does the following:
- Adds some unit tests for the `task` module. There are still some missing, however:
    - The evaluation test classes: `TestEvaluate`, `TestEvaluatePytorchJax` and `TestEvaluatePytorchJaxSingleIteration`
    - The `TestComputeMetrics` test class
    - The model loading test classes: `TestLoadModel`, `TestLoadPytorchModel` and `TestLoadSpacyModel`
- Create global `task_config` fixture, to be used in all tests
- Change the name of "text classification" to "sequence classification", to be consistent with the Hugging Face terminology
- Changed the `Task._check_supertask` to not having to insert the search string manually, and is instead derived from the supertask
- Changed default testing cache to the usual `.aiai_cache`
- Commented out the `NER` task config, to make sure it is not tested (and used) via the `get_all_task_configs` function

This does not close the associated task, but finishes the first task mentioned in that issue.
